src/sparseml/exporters/transforms/kv_cache/position_embeddings_adjustment_codegen.py

```
@@ -0,0 +1,450 @@
+        ) = self.find_nodes_to_update(model)
+        # Step 1: Add a new input to the model that will be used to pass the positions
+        model = self.add_positions_input(model)
+        # Step 2: Update the fixed positional embeddings
+        model = self.update_fixed_position_embeddings(model, position_embeddings_nodes)
+        # Step 3 Update the rotary positional embeddings
+        model = self.update_rotary_position_embeddings(
+            model, rotary_position_embedding_nodes
+        )
+
+        return model
+
+    def find_nodes_to_update(
+        self, model: ModelProto
+    ) -> Tuple[List[NodeProto], List[NodeProto]]:
+        """
+        Find all the nodes that need to be updated. For efficiency, all the nodes
+        are being found in one pass through the models' nodes.
+        """
+        position_embeddings_nodes, rotary_position_embedding_nodes = [], []
+        for node in model.graph.node:
+            # find all the nodes in every attention block that compute
+            # the fixed positional embeddings
+            if node.op_type == "Range":
+                if re.search(_POSITION_EMBEDDING_PATTERN, node.name):
+                    position_embeddings_nodes.append(node)
+            # find all the nodes in every attention block that compute
+            # the rotary positional embeddings
+            elif node.op_type == "Slice":
+                if re.search(_ROTARY_POSITION_EMBEDDING_PATTERN, node.name):
+                    rotary_position_embedding_nodes.append(node)
+
+        pattern_not_found = None
+        if not rotary_position_embedding_nodes:
+            pattern_not_found = _POSITION_EMBEDDING_PATTERN
+        if not position_embeddings_nodes:
+            pattern_not_found = _ROTARY_POSITION_EMBEDDING_PATTERN
+        if pattern_not_found:
+            raise RuntimeError(
+                f"Unable to find nodes with that match the expected pattern {pattern_not_found}"
+            )
```

why do we raise here instead of continuing?